[Sankar Ram Subramanian] iP

marraknas opened the pull request:
### **R2D2**

> Beep Bloop Blop Bleep Boop - R2D2

_Tired of finding to do lists all over the internet? Want to have an assistant who can help you with it? Fear not, R2D2 is here to help you!_

R2D2 has:

- a smooth **text based** system that is able to read your prompts quickly
- a **flat learning curve** when trying to learn how to use the chatbot!
- a super fast processing speed 🥳
- a very cute face 🥺 (see [here](https://www.google.com/url?sa=i&url=https%3A%2F%2Fscreenrant.com%2Fstar-wars-r2-d2-cutest-moments%2F&psig=AOvVaw2nS0xSKPI7Q5Y-pM4q1XDA&ust=1725604100366000&source=images&cd=vfe&opi=89978449&ved=0CBcQjhxqFwoTCPjuz62Wq4gDFQAAAAAdAAAAABAK))

_Steps to use:_

1. Add tasks through a variety of commands provided to you
2. Do the tasks and tell R2D2 to mark them as done
3. Delete tasks for a fresh day!

_Features:_
- [X]  Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class R2D2 { 
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

marraknas committed:
Rename, Greet, Exit

marraknas committed:
added Echo

marraknas committed:
Added list feature

marraknas committed:
added checkbox feature

marraknas committed:
added ToDos, Events, Deadlines

marraknas committed:
automated text ui testing

marraknas committed:
handled 2 errors

marraknas committed:
added deleting feature

marraknas committed:
added more oop

marraknas committed:
added save feature

marraknas committed:
added LocalDateTime parsing

wes-w-z-h commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,28 @@
+        DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern("d/M/yyyy HHmm");
+        return "D | " + (isDone ? "1" : "0") + " | " + description + " | " + this.date.format(dateTimeFormatter);
+    }
+    public String formatBy() {
```
Perhaps consider a more descriptive name, first time readers may not know what is by.

wes-w-z-h commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,28 @@
+            throw new BuzzException("WRONG!!! The date and time format should be d/M/yyyy HHmm (e.g., 5/10/2024 0500).");
+        }
+    }
+    public String toFileFormat() {
```
Nice method name!

marraknas committed:
Added classes to refactor code

wes-w-z-h commented on src/main/java/Deadline.java:
Missing javadocs!

wes-w-z-h commented on src/main/java/BuzzException.java:
Missing javadocs!

wes-w-z-h commented on src/main/java/Event.java:
Missing javadocs!

wes-w-z-h commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,34 @@
+import java.time.format.DateTimeParseException;
+
+public class Event extends Task {
+    protected LocalDateTime startDate;
```
Just curious, how did u decide on the visibility of these fields?

wes-w-z-h commented on src/main/java/InputHandler.java:
Missing javadocs!

wes-w-z-h commented on src/main/java/R2D2.java:
Missing javadocs!

wes-w-z-h commented on src/main/java/Task.java:
Missing javadocs!

wes-w-z-h commented on src/main/java/Todo.java:
Missing javadocs!

wes-w-z-h reviewed (commented):
lgtm! Except some files which require javadocs. See [here](https://se-education.org/guides/tutorials/checkstyle.html) for how to configure checkstyle.

marraknas committed:
added packages

marraknas committed:
Merge commit '81a9c537afe8492675eb253968939364d65db091'

marraknas committed:
added JUnit tests

marraknas committed:
packaged into a jar file

marraknas committed:
added Javadocs to most of the methods and classes

marraknas committed:
followed coding standards

marraknas committed:
added the find feature

marraknas committed:
Merge tag 'A-CodingStandard'

marraknas committed:
Merge branch 'branch-Level-9'

marraknas committed:
revised coding style violations

tanwartushar commented on src/main/java/swbot/Event.java:
```diff
@@ -0,0 +1,74 @@
+    public Event(String description, String startDate, String endDate) throws BuzzException {
+        super(description);
+        try {
+            DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern("d/M/yyyy HHmm");
```
You could consider another, more apt variable name; just like @wes-w-z-h pointed out :)

tanwartushar commented on src/main/java/swbot/Event.java:
```diff
@@ -0,0 +1,74 @@
+     * @return String that is in a readable format for the user
+     */
+    public String formatEnd() {
+        DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern("MMM dd yyyy, hh:mm a");
```
`dateTimeFormatter` is declared multiple times, could perhaps do so once as an instance variable and change its value when needed?

tanwartushar commented on src/main/java/swbot/InputHandler.java:
```diff
@@ -0,0 +1,206 @@
+        } else if (input.startsWith("find")) {
+            this.findHandle(input);
+            this.storage.saveTasks(data);
+        } else {
```
Instead of using `if-else` statements, could we consider using a `switch` statement to improve readability?

tanwartushar commented on src/main/java/swbot/InputHandler.java:
```diff
@@ -0,0 +1,206 @@
+    public void deadlineHandle(String input) throws BuzzException {
+        String[] parts = input.split("/by");
+        if (parts.length < 2 || parts[0].length() <= 9) {
+            throw new BuzzException("NOOO! Description is empty *crash*");
```
Cool statements! Work well with the theme 😁

tanwartushar commented on src/main/java/swbot/Task.java:
```diff
@@ -0,0 +1,51 @@
+package swbot;
```
Perhaps refactor `Task` and other related classes like `Deadline`, `Event` and `ToDo` into a `swbot.task` package?

tanwartushar reviewed (commented):
Looks great! A few minor adjustments and some refactoring might enhance clarity.

marraknas committed:
Changed build.gradle

marraknas committed:
Added Main.java

marraknas committed:
Added MainWindow.java

marraknas committed:
Added Launcher and DialogBox

marraknas committed:
Added resources for JavaFX

marraknas committed:
Refactored R2D2.java

marraknas committed:
Refactored InputHandler.java

marraknas committed:
Added GUI to R2D2

marraknas committed:
Add GUI in branch

marraknas committed:
Add assert statements

There are several places where assertions are important in the code to
ensure that the code does not face any runtime errors.

The assertions added currently makes sure that the task description
provided by the user is not empty or if it is null.

marraknas committed:
Refactor code to improve code quality

Most of the code for Storage.java has a lot of arrowhead style code and
nesting.

The method loadTasks() is also really long leading to a lot of LoC.

By extracting out the behaviour of a few lines of code in loadTasks(),
we can reduce the margin for error and helps debugging the code easier
as well.

Let's extract out a few behaviours in loadTasks() to other methods to
make this easier to read.

marraknas committed:
Merge pull request #1 from marraknas/branch-A-Assertions

Add assert statements

There are several places where assertions are important in the code to ensure that the code does not face any runtime errors.

The assertions added currently makes sure that the task description provided by the user is not empty or if it is null.

marraknas committed:
Merge branch 'master' into branch-A-CodeQuality

marraknas committed:
Merge pull request #2 from marraknas/branch-A-CodeQuality

Refactor code to improve code quality

Most of the code for Storage.java has a lot of arrowhead style code and nesting.

The method loadTasks() is also really long leading to a lot of LoC.

By extracting out the behaviour of a few lines of code in loadTasks(), we can reduce the margin for error and helps debugging the code easier as well.

Let's extract out a few behaviours in loadTasks() to other methods to make this easier to read.

marraknas committed:
Add gradle.yml for continuous integration

GitHub Actions will run the workflow (as per the .yml file) every time
certain project events are triggered (e.g., when a PR is updated, or the
master branch is updated).

marraknas committed:
Revert "Add gradle.yml for continuous integration"

This reverts commit b252754809bd3476215113513ce9761e97d3dd4a.

marraknas committed:
Revert "Revert "Add gradle.yml for continuous integration""

This reverts commit 7b165207da054a004abd59b882e8c3be41e7289f.

marraknas committed:
Add gradle.yml

marraknas committed:
Merge branch 'branch-A-CI'

marraknas committed:
Remove unnecessary folder

marraknas committed:
Add C-Help extension

Some users might have a hard time knowing what are the commands
available to use in the chatbot.

Having a help command to display a help page can assist users to know
what commands they can ask the chatbot to do.

Let's
* update the InputHandler to be able to handle "help" command when typed
* add a helpHandle() method to be able to display the list of commands
available in the chatbot.

marraknas committed:
Add more packages

It is hard to look through each file and see what it is for.

Sorting each class into its respective package can help to manage access
easier and makes searching/usage of classes much more easier.

Let's,
* package ui related files into one package
* package task related files into one package
* package all command related files into one package

marraknas committed:
Add javadocs for the classes

No API documentation in HTML format from comments in the code.

Add javadocs help to keep track of the documentation of each method
and different attributes, helping to create good API documentation
later.

Let's,
* add javadocs for most of the methods describing the method

marraknas committed:
Refactor code for InputHandler.java

Let's,
* follow SLAP and keep code to 30 LoC

marraknas committed:
Fix coding standards

Checkstyle was not maintained and code had a lot of errors relating to
checkstyle problems.

Let's,
* fix some errors and clean up the code in general

marraknas committed:
Remove I/O redirection test

Let's,
* remove the last 3 redirection tests since they are not applicable

marraknas committed:
Add JUnit tests

Let's,
* Test the methods more rigorously with more unit tests in place
* extensively cover all classes and their methods

marraknas committed:
Handle bugs

Let's,
* parse the event command gracefully
* squashed bugs related to event command

marraknas committed:
Update README.md

marraknas committed:
Update README.md

marraknas committed:
Modify Ui.png

marraknas committed:
Update README.md

marraknas committed:
Fix bugs

marraknas committed:
Handle bugs related to saving tasks

marraknas committed:
Fix input parsing bug

marraknas committed:
Update README.md
